altapay.php

```diff
@@ -0,0 +1,2353 @@
+        $helper->shopLinkType = '';
+        $helper->simple_header = false;
+        $helper->actions = array('edit');
+        $helper->identifier = 'id_terminal';
+        $helper->position_identifier = 'position';
+        $helper->show_toolbar = true;
+        $helper->toolbar_btn = array(
+            'new' => array(
+                'href' => AdminController::$currentIndex . '&configure=' . $this->name . '&add' . $this->name . '&token='
+                    . Tools::getAdminTokenLite('AdminModules'),
+                'desc' => $this->l('Add new')
+            )
+        );
+        $helper->title = 'Terminals';
+        $helper->table = 'altapay_terminals';
+        $helper->token = Tools::getAdminTokenLite('AdminModules');
+        $helper->currentIndex = AdminController::$currentIndex . '&configure=' . $this->name;
+        $helper->orderBy = 'id_terminal';
+        $helper->orderWay = 'ASC';
+        $content = Terminal::getTerminals();
+
+        return $helper->generateList($content, $fields_list);
+    }
+
+    /**
+     * Validate merchant details form
+     */
+    private function postValidation()
+    {
+        if (Tools::isSubmit('btnSubmit')) {
+            if (!Tools::getValue('ALTAPAY_USERNAME')) {
+                $this->postErrors[] = $this->l('API username is required');
+            } elseif (!Tools::getValue('ALTAPAY_URL')) {
+                $this->postErrors[] = $this->l('API URL is required');
+            }
+            if (!filter_var(Tools::getValue('ALTAPAY_URL'), FILTER_VALIDATE_URL)) {
+                $this->postErrors[] = $this->l('Incorrect format for the API URL - Use "https://paymentURL"');
+            }
+        }
+    }
+
+    /**
+     * Method for saving gateway configuration details in plugin settings
+     */
+    private function postProcess()
+    {
+        if (Tools::isSubmit('btnSubmit')) {
+            Configuration::updateValue('ALTAPAY_USERNAME', Tools::getValue('ALTAPAY_USERNAME'));
+            $urlPath = preg_replace('/\s+/', '', Tools::getValue('ALTAPAY_URL'));
+            if (Tools::substr($urlPath, -1) != '/') {
+                Configuration::updateValue('ALTAPAY_URL', $urlPath .= '/');
+            } elseif (Tools::substr($urlPath, -1) == '/') {
+                Configuration::updateValue('ALTAPAY_URL', $urlPath);
+            }
+            if (Tools::getValue('ALTAPAY_PASSWORD') !== "") {
+                Configuration::updateValue('ALTAPAY_PASSWORD', Tools::getValue('ALTAPAY_PASSWORD'));
+            }
+            if (Tools::getValue('AUTOCAPTURE_STATUSES') !== "") {
+                Configuration::updateValue('AUTOCAPTURE_STATUSES', serialize(Tools::getValue("AUTOCAPTURE_STATUSES")));
+            }
+        }
+        $this->Mhtml .= '<div class="alert alert-success"> ' . $this->l('Settings updated') . '</div>';
+    }
+
+    /**
+     * Displays the error that occurred in the hookActionOrderStatusUpdate method, if any.
+     *
+     * @param $params
+     * @return mixed
```

Comment:
mixed is useless and should be avoided when not explicitly true
```suggestion
     * @return bool
```